### ISSUE TITLE
Add a workspace test to prevent incompatible wasm issue happend last week

### ISSUE THREAD

**Issue by ailisp** [near/devgigsboard#35]
**Problem**
We hit a incompatible wasm compilation by rustc 1.70 issue. As a result, the incompatible wasm cannot be loaded by nearcore and we cannot call `self_upgrade` to update it. This can only be caught by deploying the contract to a node and test it on CI, before deploy to production.

**Acceptance Criteria** 
- Add a workspace test that deploy the contract and call `self_upgrade` method as sanity check.
- Add it to CI, using the **exactly same** configuration that uses for build and deploy to production

**Comment by jaswinder6991** [comment]
I am starting to learn the workspaces-rs project. I can take this up. I might take more time than expected. If this is not a priority, I want to work on this.

**Comment by frol** [comment]
Just for the context, I touched a little bit on the topic of workspaces-rs on the call working on the state migration: https://drive.google.com/file/d/182K4BPJdHZw-TjqqTvSgnnLKBs9j7tt8/view?usp=drivesdk

**Comment by jaswinder6991** [comment]
I actually got the inspiration to work on this after watching the video. I also want to start contributing to the workspaces project. So I feel like using it for a contract might be the best way to get to know it.

**Comment by jaswinder6991** [comment]
I know this issue is only for the self_upgrade functions testing but I feel like we should have a foundation to start writing workspaces tests for all upcoming functionality and we can also maybe in parallel write tests for features which are already present.

**Comment by jaswinder6991** [comment]
I wrote a simple [branch](https://github.com/jaswinder6991/neardevhub-contract/tree/setup-workspaces) to setup workspaces-rs test suits. It is not PR ready but would want to get your comments.

I have a question too. I compiled the contract with rust 1.71.0 and deployed it to testnet and it worked. I am assuming the self_upgrade won't work though. I am not sure why it would successfully deploy. Some information for my personal knowledge here.

In the test, I wrote. If the code is compiled with 1.71.0, workspaces won't even let it deploy. So if we are putting a test for rust version incompatibility, we don't really have to do self_upgrade function for sanity check. Unless there is something I am missing.
Here is error that I see on workspaces if I deploy a 1.71.0 version - 
status: Failure(ActionError(ActionError { index: Some(0), kind: FunctionCallError(CompilationError(PrepareError(Deserialization))) }))
